offline/QA/Jet/photonjetskinematics.cc

```diff
@@ -0,0 +1,184 @@
+  if (!clusterContainer)
+    {
+      std::cout << PHWHERE << "funkyCaloStuff::process_event - Fatal Error - CLUSTER_CEMC node is missing. " << std::endl;
+      return 0;
+    }
+
+  RawClusterContainer::ConstIterator clusterIter;
+  RawClusterContainer::ConstRange clusterEnd = clusterContainer->getClusters();
+
+  for (clusterIter = clusterEnd.first; clusterIter != clusterEnd.second; clusterIter++)
+    {
+      RawCluster* recoCluster = clusterIter->second;
+      CLHEP::Hep3Vector vertex(0, 0, 0);
+      CLHEP::Hep3Vector E_vec_cluster = RawClusterUtility::GetECoreVec(*recoCluster, vertex);
+
+      float clusE = E_vec_cluster.mag();
+      float clus_eta = E_vec_cluster.pseudoRapidity();
+      float clus_phi = E_vec_cluster.phi();
+      float clus_chisq = recoCluster->get_chi2();
+ 
+
+      //Filling Histograms, only taking into account E vec cluster, not reco cluster
+
+       h_emcal_cluster_chi2->Fill(clus_chisq);
+       h_emcal_cluster_energy->Fill(clusE);
+       h_emcal_cluster_eta_phi->Fill(clus_eta, clus_phi);
+       h_emcal_cluster_eta->Fill(clus_eta);  // 1D eta plot
+       h_emcal_cluster_phi->Fill(clus_phi);  // 1D phi plot
+
+       // std::cout << "E vec phi = " << clus_phi << std::endl;
+
+       // std::cout << "E vec energy = " << clusE << std::endl;
+
+       // std::cout << "E vec chi2  = " << clus_chisq << std::endl;
+
+       // std::cout << "E vec eta = " << clus_eta << std::endl;
+         
+  }
+
+  // std::cout << "photonjetskinematics::process_event(PHCompositeNode *topNode) Processing Event" << std::endl;
+  return Fun4AllReturnCodes::EVENT_OK;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::ResetEvent(PHCompositeNode *topNode)
+{
+  // std::cout << "photonjetskinematics::ResetEvent(PHCompositeNode *topNode) Resetting internal structures, prepare for next event" << std::endl;
+  return Fun4AllReturnCodes::EVENT_OK;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::EndRun(const int runnumber)
+{
+  std::cout << "photonjetskinematics::EndRun(const int runnumber) Ending Run for Run " << runnumber << std::endl;
+  return Fun4AllReturnCodes::EVENT_OK;
+}
+
+//____________________________________________________________________________..
+int photonjetskinematics::End(PHCompositeNode *topNode)
+{
+  std::cout << "photonjetskinematics::End - Output to " << outfilename << std::endl;
+  PHTFileServer::get().cd(outfilename);
+
+  //outfile->cd("photonjetskinematics");
+
+  //Outputting the histograms
+  h_emcal_cluster_eta_phi->Write("h_emcal_cluster_eta_phi");
+  h_emcal_cluster_energy->Write("h_emcal_cluster_energy");
+  h_emcal_cluster_chi2->Write("h_emcal_cluster_chi2");
+  h_emcal_cluster_eta->Write("h_emcal_cluster_eta");  // Save 1D eta plot
+  h_emcal_cluster_phi->Write("h_emcal_cluster_phi");  // Save 1D phi plot
```

Comment:
Here we instead register the output histograms with the manager rather than save them to a particular file.
```suggestion
  manager->registerHisto(h_emcal_cluster_eta_phi);
  manager->registerHisto(h_emcal_cluster_energy);
  manager->registerHisto(h_emcal_cluster_chi2);
  manager->registerHisto(h_emcal_cluster_eta);  // Save 1D eta plot
  manager->registerHisto(h_emcal_cluster_phi);  // Save 1D phi plot
```